test/vmtests/vmtests/utils/imagecreator.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+from pathlib import Path
+from typing import List
+
+from . import local_client
+
+
+# Run the imagecreator binary tool.
+def run_image_creator(
+    image_creator_binary_path: Path,
+    rpm_sources: List[Path],
+    tools_tar: Path,
+    config_path: Path,
+    output_image_format: str,
+    output_image_path: Path,
+    build_dir: Path,
+) -> None:
+    # Validate required files exist before starting
+    if not image_creator_binary_path.exists():
```

Comment:
Why are you prechecking for these conditions? Why not just let the error be produced when `local_client.run` is called?